crates/rnote-cli/src/export.rs

```diff
@@ -0,0 +1,715 @@
+                                Bounds::Inside => engine.store.keys_sorted_chrono_in_bounds(aabb),
+                                Bounds::Intersecting => {
+                                    engine.store.keys_unordered_intersecting_bounds(aabb)
+                                }
+                            },
+                            "No strokes in given rectangle",
+                        )
+                    } else if selection.all {
+                        (engine.store.stroke_keys_unordered(), "Document is empty")
+                    } else {
+                        // Clap should make sure eighter of them are used
+                        return Err(anyhow::anyhow!(" --all or --rect required"));
+                    };
+                    if strokes.is_empty() {
+                        return Err(anyhow::anyhow!("{err_msg}"));
+                    }
+                    engine.store.set_selected_keys(&strokes, true);
+                    engine
+                        .export_selection(None)
+                        .await??
+                        .context("No strokes selected")?
+                }
+                ExportCommands::Doc { .. } => engine.export_doc(export_file_name, None).await??,
+                ExportCommands::DocPages { .. } => unreachable!(),
+            };
+            let mut fh = File::create(output_file).await?;
+            fh.write_all(&export_bytes).await?;
+            fh.sync_all().await?;
+        }
+        ExportCommands::DocPages {
+            output_dir,
+            output_file_stem,
+            output_format,
+            ..
+        } => {
+            validators::path_is_dir(output_dir)?;
+            // The output file cannnot be set with this subcommand
+            drop(output_file);
+
+            let export_bytes = engine.export_doc_pages(None).await??;
+            let out_ext = DocPagesExportFormat::from(output_format).file_ext();
+            let output_file_stem = match output_file_stem {
+                Some(o) => o.clone(),
+                None => match rnote_file.file_stem() {
+                    Some(stem) => stem.to_string_lossy().to_string(),
+                    None => {
+                        return Err(anyhow::anyhow!(
+                            "Failed to generate output_file_stem from rnote_file"
+                        ))
+                    }
+                },
+            };
+            for (i, bytes) in export_bytes.into_iter().enumerate() {
+                export_doc_page(
+                    i,
+                    output_dir,
+                    &output_file_stem,
+                    &out_ext,
+                    &bytes,
+                    on_conflict,
+                )
+                .await
+                .context(format!(
+                    "Failed to export page {i} of document {}",
+                    rnote_file.display()
+                ))?
+            }
+        }
+    };
+    Ok(())
+}
+
+async fn export_doc_page(
```

Comment:
This function name is wrong, the export happened already earlier. Also the user needs to be asked prior to actually exporting.